src/app/clusters/network-commissioning/NetworkCommissioningLogic.h

```diff
@@ -0,0 +1,242 @@
+    std::optional<DataModel::ActionReturnStatus>
+    HandleConnectNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                         const NetworkCommissioning::Commands::ConnectNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus>
+    HandleReorderNetwork(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                         const NetworkCommissioning::Commands::ReorderNetwork::DecodableType & req);
+    std::optional<DataModel::ActionReturnStatus> HandleNonConcurrentConnectNetwork();
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+    std::optional<DataModel::ActionReturnStatus>
+    HandleQueryIdentity(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                        const NetworkCommissioning::Commands::QueryIdentity::DecodableType & req);
+
+    std::optional<DataModel::ActionReturnStatus>
+    HandleAddOrUpdateWiFiNetworkWithPDC(CommandHandler & handler, const ConcreteCommandPath & commandPath,
+                                        const NetworkCommissioning::Commands::AddOrUpdateWiFiNetwork::DecodableType & req);
+#endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+
+    // Attribute handling
+    CHIP_ERROR SetInterfaceEnabled(bool enabled) { return mpBaseDriver->SetEnabled(enabled); }
+    uint8_t GetMaxNetworks() const { return mpBaseDriver->GetMaxNetworks(); }
+    uint8_t GetScanNetworkTimeoutSeconds() const
+    {
+        return mpWirelessDriver != nullptr ? mpWirelessDriver->GetScanNetworkTimeoutSeconds() : 0;
+    }
+    uint8_t GetConnectNetworkTimeoutSeconds() const
+    {
+        return mpWirelessDriver != nullptr ? mpWirelessDriver->GetConnectNetworkTimeoutSeconds() : 0;
+    }
+    bool GetInterfaceEnabled() const { return mpBaseDriver->GetEnabled(); }
+    NetworkCommissioning::Attributes::LastNetworkingStatus::TypeInfo::Type GetLastNetworkingStatus() const
+    {
+        return mLastNetworkingStatusValue;
+    }
+    NetworkCommissioning::Attributes::LastConnectErrorValue::TypeInfo::Type GetLastConnectErrorValue() const
+    {
+        return mLastConnectErrorValue;
+    }
+    ByteSpan GetLastNetworkID() const { return { mLastNetworkID, mLastNetworkIDLen }; }
+
+    uint16_t GetThreadVersion() const
+    {
+#if (CHIP_DEVICE_CONFIG_ENABLE_THREAD)
+        if (mFeatureFlags.Has(NetworkCommissioning::Feature::kThreadNetworkInterface))
+        {
+            return mpDriver.Get<DeviceLayer::NetworkCommissioning::ThreadDriver *>()->GetThreadVersion();
+        }
+#endif
+        return 0;
+    }
+
+    BitMask<chip::DeviceLayer::NetworkCommissioning::ThreadCapabilities> GetThreadCapabilities() const
+    {
+#if (CHIP_DEVICE_CONFIG_ENABLE_THREAD)
+        if (mFeatureFlags.Has(NetworkCommissioning::Feature::kThreadNetworkInterface))
+        {
+            return mpDriver.Get<chip::DeviceLayer::NetworkCommissioning::ThreadDriver *>()->GetSupportedThreadFeatures();
```

Comment:
Why all the inconsistent namespacing?  It's `chip::DeviceLayer::NetworkCommissioning::ThreadDriver` but `DeviceLayer::NetworkCommissioning::ThreadDriver` 11 lines above...  In the original code it was just `ThreadDriver`, which was vastly better.

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/39721